scripts/util.py

```diff
@@ -4,27 +4,36 @@
 from cassandra.cqlengine.query import Token
 
 from scrapi.database import _manager
-from scrapi.processing.cassandra import DocumentModel
+from scrapi.processing.cassandra import DocumentModel, DocumentModelV2
 
 _manager.setup()
 logger = logging.getLogger(__name__)
 
 
-def documents(*sources):
-    q = DocumentModel.objects.timeout(500).allow_filtering().all().limit(1000)
-    querysets = (q.filter(source=source) for source in sources) if sources else [q]
-    for query in querysets:
-        page = try_forever(list, query)
-        while len(page) > 0:
-            for doc in page:
-                yield doc
-            page = try_forever(next_page, query, page)
+def ModelIteratorFactory(model, next_page):
+    def model_iterator(*sources):
+        q = model.objects.timeout(500).allow_filtering().all().limit(1000)
+        querysets = (q.filter(source=source) for source in sources) if sources else [q]
+        for query in querysets:
+            page = try_forever(list, query)
+            while len(page) > 0:
+                for doc in page:
+                    yield doc
+                page = try_forever(next_page, query, page)
```

Comment:
so if you notice the factory takes a model and a next page function. So the actual iterators, ```documents_v1``` or ```documents_v2``` are created with ```DocumentModel``` and ```next_page_v1``` or ```DocumentModelV2``` and ```next_page_v2```. So the original model will be with ```next_page_v1```, and the new model will be with ```next_page_v2```